### PR TITLE
Add support for unix sockets

### PR DESCRIPTION
**Pull request by bigs**
Pretty sure I handled this in a way that won't get us into trouble down the line. Would love some eyes on this.

### CODE REVIEW

**Comment by bigs** [comment]
Rebased.

**Comment by bigs** [comment]
how do you mean re: dialargs @vyzo? happy to update

**Comment by vyzo** [comment]
Code is fine, I am just not sure of its utility :)

**Comment by Stebalien** [comment]
LGTM but, FYI, there's a bit of an unresolved issue with unix addresses: they don't compose. That is, we can't write `/unix/a/b/c/p2p/Qm...` and we still haven't found a good way around this.

File: convert.go [hunk 1]
```diff
@@ -137,6 +140,10 @@ func DialArgs(m ma.Multiaddr) (string, string, error) {
 				hostname = true
 				ip = c.Value()
 				return true
+			case ma.P_UNIX:
```
Comment (lanzafame) [inline convert.go]: function doc needs updating with this change.

Comment (bigs) [reply convert.go]: done!

**Pull request merged**